### PR TITLE
[WIP] Various Undefined variable fixes

### PR DESCRIPTION
**Pull request by sivel**
##### SUMMARY

Currently we are ignoring `undefined-variable` in pylint.  We probably shouldn't do this long term.

This is an initial set of fixes that I could perform. There are a few other places where:

1. The code where the undefined var is located isn't used anywhere
1. The code is a bit complicated (double lambda, cascaded var inspection) and I haven't determined the proper outcome
1. There is really no way to make the code work without understanding how it's supposed to work, in the places the code is completely broken now.

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
various

##### ANSIBLE VERSION

```
devel
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Comment by sivel** [comment]
@jctanner in `module_utils/vmware.py` there are 2 functions, that aren't used anywhere, and don't have access to the vars they use.  Do you think it's OK, to just remove those functions?

```
ansible/module_utils/vmware.py:385: [E0602(undefined-variable), fetch_file_from_guest] Undefined variable 'fetch_url'
ansible/module_utils/vmware.py:385: [E0602(undefined-variable), fetch_file_from_guest] Undefined variable 'self'
ansible/module_utils/vmware.py:430: [E0602(undefined-variable), push_file_to_guest] Undefined variable 'os'
ansible/module_utils/vmware.py:446: [E0602(undefined-variable), push_file_to_guest] Undefined variable 'fetch_url'
ansible/module_utils/vmware.py:446: [E0602(undefined-variable), push_file_to_guest] Undefined variable 'self'
```

**Comment by Akasurde** [comment]
@sivel I have a couple fixes for that in this PR #25288

File: lib/ansible/modules/cloud/azure/azure.py [hunk 1]
```diff
@@ -305,7 +306,7 @@ def _handle_timeout(signum, frame):
                     azure.delete_disk(disk.name, True)
                     disk_names.remove(disk_name)
     except AzureException as e:
-        module.fail_json(msg="failed to get or delete disk %s, error was: %s" % (disk_name, str(e)))
+        raise AzureException("failed to get or delete disk %s, error was: %s" % (disk_name, str(e)))
```
Comment (bcoca) [inline lib/ansible/modules/cloud/azure/azure.py:309]: this is  not a fix, module should always exist gracefully (not in exception) .. the str() should be changed to_native though

**Review by bcoca — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/network/panos/panos_restart.py [hunk 2]
```diff
@@ -104,10 +104,9 @@ def main():
     except Exception:
         x = sys.exc_info()[1]
         if 'succeeded' in str(x):
-            module.exit_json(changed=True, msg=str(msg))
```
Comment (bcoca) [inline lib/ansible/modules/network/panos/panos_restart.py:107]: str should probably be converted to_native here also 

@abadger

**Review by bcoca — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/cloud/azure/azure.py (hunk 1 above)
Comment (sivel) [reply lib/ansible/modules/cloud/azure/azure.py:309]: The module later catches this exception already and converts the exception for use with `fail_json`

File: lib/ansible/plugins/strategy/debug.py [hunk 3]
```diff
@@ -177,6 +177,5 @@ def execute(self, args):
     def default(self, line):
         try:
             self.execute(line)
-            display.display(pprint.pformat(result))
```
Comment (bcoca) [inline lib/ansible/plugins/strategy/debug.py:180]: doesn't this break functionality for debug strategy?

**Review by bcoca — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/network/panos/panos_restart.py (hunk 2 above)
Comment (sivel) [reply lib/ansible/modules/network/panos/panos_restart.py:107]: I get that.  And I can make these changes.  However my goal here is to do very targeted changes to resolve the undefined errors, not necessarily do py3 compat.

File: lib/ansible/plugins/strategy/debug.py (hunk 3 above)
Comment (sivel) [reply lib/ansible/plugins/strategy/debug.py:180]: No, that var is completely undefined at that point, so this is already broken. I also couldn't find any useful information from the called methods that I could actually return.

File: lib/ansible/modules/cloud/azure/azure.py (hunk 1 above)
Comment (sivel) [reply lib/ansible/modules/cloud/azure/azure.py:309]: https://github.com/sivel/ansible/blob/bd1779292ff1c11bdc7e98c916533bf28be4b00a/lib/ansible/modules/cloud/azure/azure.py#L500

**Pull request closed without merging**